src/org/opendatakit/briefcase/ui/export/components/ConfigurationPanelForm.java

```diff
@@ -89,26 +88,19 @@
         buildPemFileDialog().choose().ifPresent(file -> setPemFile(Paths.get(file.toURI())))
     );
     pemFileClearButton.addActionListener(__ -> clearPemFile());
+
     startDatePicker.addDateChangeListener(event -> {
-      if (!isDateRangeValid()) {
-        showError(MessageStrings.INVALID_DATE_RANGE_MESSAGE, "Export configuration error");
-        startDatePicker.clear();
-      } else
-        onSelectStartDateCallbacks.forEach(consumer -> consumer.accept(extractDate(event)));
-    });
-    endDatePicker.addDateChangeListener(event -> {
-      if (!isDateRangeValid()) {
-        showError(MessageStrings.INVALID_DATE_RANGE_MESSAGE, "Export configuration error");
+
+      if (event.getNewDate() != null && isAfterEndDate(event.getNewDate()))
         endDatePicker.clear();
-      } else
-        onSelectEndDateCallbacks.forEach(consumer -> consumer.accept(extractDate(event)));
+
+      if (event.getNewDate() != null)
+        endDatePicker.getSettings().setDateRangeLimits(event.getNewDate().plusDays(1), null);
+
+      onSelectStartDateCallbacks.forEach(consumer -> consumer.accept(event.getNewDate()));
     });
-  }
 
-  private boolean isDateRangeValid() {
-    LocalDate startDate = startDatePicker.getDate();
-    LocalDate endDate = endDatePicker.getDate();
-    return startDate == null || endDate == null || startDate.isBefore(endDate);
+    endDatePicker.addDateChangeListener(event -> onSelectEndDateCallbacks.forEach(consumer -> consumer.accept(event.getNewDate())));
```

Comment:
On this change listener we're missing a `startDatePicker.getSettings().setDateRangeLimits(null, event.getNewDate());` to apply restrictions to the start datepicker.